frontend/src/utils/filterHelpers.js

```diff
@@ -322,3 +328,40 @@ export function parseFiltersToPatch(params) {
     return acc;
   }, []);
 }
+
+const cleanupParameter = ({ parameterName, value, valueTo, defaultValue }) => {
+  value = value === null && defaultValue ? defaultValue : value;
+
+  return {
+    parameterName,
+    value:
+      value &&
+      value.values &&
+      Array.isArray(value.values) &&
+      value.values.length === 0
+        ? [] // case when facets gets cleared
+        : value,
+    valueTo,
+  };
+};
+
+export function cleanupFilter({ filterId, parameters }) {
```

Comment:
I would rename this method to `prepareFilterForBackend` to explicitly say what it does.